lib/components/user/nav-login-button.js

```diff
@@ -0,0 +1,124 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import {
+  MenuItem,
+  NavDropdown,
+  NavItem
+} from 'react-bootstrap'
+
+/**
+ * This component displays the sign-in status in the nav bar.
+ * - When a user is not logged in: display 'Sign In' as a link or button.
+ * - When a user is logged in, display an 'avatar' (retrieved from the profile prop)
+ *   and a dropdown button so the user can access more options.
+ */
+export default class NavLoginButton extends Component {
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    links: PropTypes.arrayOf(PropTypes.shape({
+      target: PropTypes.string,
+      text: PropTypes.string.isRequired,
+      url: PropTypes.string.isRequired
+    })),
+    onSignInClick: PropTypes.func.isRequired,
+    onSignOutClick: PropTypes.func.isRequired,
+    profile: PropTypes.shape({
+      email: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      nickname: PropTypes.string,
+      picture: PropTypes.string
+    })
+  }
+
+  static defaultProps = {
+    links: null,
+    profile: null
+  }
+
+  render () {
+    const {
+      className,
+      id,
+      links,
+      onSignInClick,
+      onSignOutClick,
+      profile,
+      style
+    } = this.props
+
+    const commonProps = {
+      className,
+      id,
+      style
+    }
+
+    /**
+      * Override bootstrap's defaults, especially for nav links.
+      * HACK: Because we don't want to use the rendered <NavDropDown> inside a <Nav> while in mobile mode
+      * (otherwise, the menu items get displayed full screen width, which is undesirable),
+      * the CSS rules below let us fit the sign-in component to the top-right corner
+      * and add some of the default visuals provided by react-bootstrap's <Nav> while in mobile mode.
+      * We change the link color to white for consistency with the AppMenu component.
+      */
+    const styleOverrides = (
```

Comment:
@binh-dam-ibigroup, I would prefer for this css to be in a separate css file, similar to how we handle the other pieces of this application. Any reason why it's inline with the js?